Snowflake.Data/Logger/EasyLoggerManager.cs

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System;
+using System.IO;
+using log4net;
+using log4net.Appender;
+using log4net.Layout;
+using Snowflake.Data.Configuration;
+
+namespace Snowflake.Data.Log
+{
+    internal class EasyLoggerManager
+    {
+        public static readonly EasyLoggerManager Instance = new EasyLoggerManager();
+
+        private readonly object _lockForExclusiveConfigure = new object();
+
+        private const string AppenderPrefix = "SFEasyLogging";
+
+        private readonly EasyLoggingLevelMapper _levelMapper = EasyLoggingLevelMapper.Instance;
+        
+        public virtual void ReconfigureEasyLogging(EasyLoggingLogLevel easyLoggingLogLevel, string logsPath)
+        {
+            var log4netLevel = _levelMapper.ToLog4NetLevel(easyLoggingLogLevel);
+            lock (_lockForExclusiveConfigure)
+            {
+                var repository = (log4net.Repository.Hierarchy.Hierarchy)LogManager.GetRepository();
+                var rootLogger = (log4net.Repository.Hierarchy.Logger)repository.GetLogger("Snowflake.Data");
+                rootLogger.Level = log4netLevel;
+                var appender = string.Equals(logsPath, "STDOUT", StringComparison.OrdinalIgnoreCase)
+                    ? AddConsoleAppender(rootLogger)
+                    : AddRollingFileAppender(rootLogger, logsPath);
+                RemoveOtherAppendersForEasyLogging(rootLogger, appender);
+                repository.RaiseConfigurationChanged(EventArgs.Empty);
+            }
+        }
+        
+        private static void RemoveOtherAppendersForEasyLogging(log4net.Repository.Hierarchy.Logger logger, IAppender appender)
+        {
+            var existingAppenders = logger.Appenders.ToArray();
+            foreach (var existingAppender in existingAppenders)
+            {
+                if (IsAppenderCreatedForEasyLogging(existingAppender) && existingAppender != appender)
+                {
+                    logger.RemoveAppender(existingAppender);
+                }
+            }
+        }
+
+        private static IAppender AddRollingFileAppender(log4net.Repository.Hierarchy.Logger logger,
+            string directoryPath)
+        {
+            var patternLayout = PatternLayout();
+            var randomFileName = $"snowflake_dotnet_{Path.GetRandomFileName()}";
```

Comment:
Do you think that appending the timestamp of the application run/appender addition to the file name will be a good idea?

Reply:
it is not so easy... First of all in the jdbc driver there is also only something unique in the name, not a timestamp. 

Let's say we have such log files:
- snowflake_dotnet_abcxyz.log
- snowflake_dotnet_abcxyz_1.log
- snowflake_dotnet_abcxyz_2.log

If I would like to add a timestamp to a file pattern I would add the timestamp to all of these files which could be confusing. 

If I would change the rolling style from based on size to composite (based on size and date) and specify date pattern including time with - let's say - a minute precision, then we would create a new file every minute - so it would be very bad.